### PR TITLE
Add: nasl-cli notus update subcommand

### PR DESCRIPTION
**Pull request by jjnicola**
**What**:
Load up Notus Advisories into redis.
It performs the signature check, the hashsum check and the upload.

Signature check is optional. It must be enabled with the command line option but also the environment variable to the gnupg keyring must be set.

Usage:
`GPGHOME=/path/to/.gnupg nasl-cli notus update --path <path-to-the-advisories> --signature-check`

Jira: SC-997

**Why**:

**How**:

**Checklist**:

- [ ] Tests
- [ ] PR merge commit message adjusted

### CODE REVIEW

File: rust/models/src/advisories.rs [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+            filename: data.filename.to_string(),
+            refs,
+            family: data.product_data.family.to_owned(),
+            name: data.adv.title.to_owned(),
+            category: "3".to_string(),
+        }
+    }
+}
+
+//impl Vulnerability {
```
Comment (nichtsfrei) [inline rust/models/src/advisories.rs:199]: can this be deleted?

File: rust/nasl-cli/src/notusupdate/mod.rs [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+                }
+            };
+
+            let signature_check = args
+                .get_one::<bool>("signature-check")
+                .cloned()
+                .unwrap_or(false);
+
+            let dispatcher =
+                redis_storage::NvtDispatcher::as_dispatcher(&redis, NOTUSUPDATE_SELECTOR)
```
Comment (nichtsfrei) [inline rust/nasl-cli/src/notusupdate/mod.rs:59]: maybe the NvtDispatcher of redis_storage should be renamed?

File: rust/nasl-cli/src/notusupdate/update.rs~ [hunk 3]
```diff
@@ -0,0 +1,59 @@
+// SPDX-FileCopyrightText: 2023 Greenbone AG
```
Comment (nichtsfrei) [inline rust/nasl-cli/src/notusupdate/update.rs~:1]: seems like a swap file?

File: rust/notus/src/update/#mod.rs# [hunk 4]
```diff
@@ -0,0 +1,69 @@
+// SPDX-FileCopyrightText: 2023 Greenbone AG
```
Comment (nichtsfrei) [inline rust/notus/src/update/#mod.rs#:1]: seeme like a swap file?

File: rust/models/src/advisories.rs (hunk 1 above)
Comment (jjnicola) [reply rust/models/src/advisories.rs]: ups.. that was my try of serializer for vulnerabilities.

File: rust/nasl-cli/src/notusupdate/mod.rs (hunk 2 above)
Comment (jjnicola) [reply rust/nasl-cli/src/notusupdate/mod.rs]: I thought about it and forgot to do it.

File: rust/notus/src/update/#mod.rs# (hunk 4 above)
Comment (jjnicola) [reply rust/notus/src/update/#mod.rs#]: ups. Also, clean up other files.

File: rust/nasl-cli/src/notusupdate/update.rs~ (hunk 3 above)
Comment (jjnicola) [reply rust/nasl-cli/src/notusupdate/update.rs~]: ups

**Comment by jjnicola** [comment]
thanks for the review @nichtsfrei

**Comment by nichtsfrei** [comment]
I have been thinking about the separation between the 'notus' and 'feed' commands. Although it is technically correct to separate them, I'm questioning whether users actually need this differentiation or if they would rather 'verify' and 'update' them in one command?

**Comment by nichtsfrei** [comment]
> I have been thinking about the separation between the 'notus' and 'feed' commands. Although it is technically correct to separate them, I'm questioning whether users actually need this differentiation or if they would rather 'verify' and 'update' them in one command?

maybe something like `nasl-cli feed update` -> notus + nasl, `nasl-cli feed --notus update` -> only notus and the same for nasl?

**Review by nichtsfrei — APPROVED** [review]

**Review by ArnoStiefvater — APPROVED** [review]

**Pull request merged**